File: app/src/main/java/com/falcon/restaurants/screens/mealdetail/MealDetailsActivity.java
package com.falcon.restaurants.screens.mealdetail;

import android.content.Intent;
import android.os.Bundle;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import com.falcon.restaurants.R;
import com.falcon.restaurants.room.meal.Meal;
import com.falcon.restaurants.screens.common.BaseActivity;
import com.falcon.restaurants.screens.common.ImageLoader;
import com.falcon.restaurants.screens.meal.MealViewModel;
import com.falcon.restaurants.utils.Constants;
import com.falcon.restaurants.utils.Logger;

import javax.inject.Inject;

public class MealDetailsActivity extends BaseActivity {

    String TAG = "MealDetailsActivity" ;

    String mealId = "" ;
    public static final String MEAL_ID = "MEAL_ID";

    @Inject public MealViewModel mealViewModel;
    @Inject public ImageLoader imageLoader;

    private ImageView meal_img;
    private TextView name_txt;
    private TextView details_txt;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        getPresentationComponent().inject(this);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_meal_details);

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        meal_img = findViewById(R.id.meal_img);
        name_txt = findViewById(R.id.name_txt);
        details_txt = findViewById(R.id.details_txt);

        mealId = getIntent().getStringExtra(MealDetailsActivity.MEAL_ID);
        Logger.log( TAG, "onCreate: mealId: " + mealId);

        if(!mealId.isEmpty()) {
            mealViewModel.getMealById(mealId, new MealViewModel.ListenerForOne() {
                @Override
                public void onQuerySuccess(Meal meal) {
                    if (meal !=null){
                        imageLoader.loadImage(meal_img, Constants.imagesMealsUrl + meal.getImageUrl());

                        name_txt.setText(meal.getName());
                        details_txt.setText(meal.getDetails());
                    }
                }
                @Override
                public void onQueryFailed(Throwable e) {
                }
            });
        }
    }

    public static void start(AppCompatActivity fromActivity, String mealId){
        Intent intent = new Intent(fromActivity, MealDetailsActivity.class);
        intent.putExtra(MealDetailsActivity.MEAL_ID, mealId );
        fromActivity.startActivity(intent);
    }

}

File: app/src/main/java/com/falcon/restaurants/screens/mealdetail/MealDetailsActivity.kt
package com.falcon.restaurants.screens.mealdetail
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import androidx.appcompat.app.AppCompatActivity
import com.falcon.restaurants.databinding.ActivityMealDetailsBinding
import com.falcon.restaurants.room.meal.Meal
import com.falcon.restaurants.screens.common.BaseActivity
import com.falcon.restaurants.screens.common.ImageLoader
import com.falcon.restaurants.screens.meal.MealViewModel
import com.falcon.restaurants.utils.Constants
import com.falcon.restaurants.utils.Logger
import javax.inject.Inject

class MealDetailsActivity : BaseActivity() {

    val TAG: String = "MealDetailsActivity"
    var mealId: String = ""
    lateinit var binding: ActivityMealDetailsBinding

    @Inject lateinit var mealViewModel: MealViewModel
    @Inject lateinit var imageLoader: ImageLoader
    @Inject lateinit var layoutInflator: LayoutInflater

    companion object {
        const val MEAL_ID: String = "MEAL_ID"
        fun start(fromActivity: AppCompatActivity, mealId: String)
        {
            val intent = Intent(fromActivity, MealDetailsActivity::class.java)
            intent.putExtra(MEAL_ID, mealId)
            fromActivity.startActivity(intent)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        presentationComponent.inject(this)
        super.onCreate(savedInstanceState)
        binding = ActivityMealDetailsBinding.inflate(layoutInflator)
        setContentView(binding.root)
        setSupportActionBar(binding.toolbar)

        if(intent.hasExtra(MEAL_ID)) {
            mealId = intent.getStringExtra(MEAL_ID)!!
        }

        Logger.log( TAG, "onCreate: mealId: $mealId")

        if(!mealId.isEmpty()) {
            mealViewModel.getMealById(mealId, object :MealViewModel.ListenerForOne {

                override fun onQuerySuccess(meal: Meal) {
                    if (meal !=null){
                        imageLoader.loadImage(binding.mealImg, Constants.imagesMealsUrl + meal.imageUrl)
                        binding.nameTxt.setText(meal.name)
                        binding.detailsTxt.setText(meal.details)
                    }
                }
                override fun onQueryFailed(e: Throwable) {
                }
            })
        }
    }
}
